tests/test.py

```diff
@@ -0,0 +1,75 @@
+import sys
+import socket
+
+HOST = '127.0.0.1'
+PORT = 7777 # The port used by the server
+
+def recv_all(sock, n):
+    b = bytearray()
+    read = 0
+    while read < n:
+        tmp = sock.recv(n-read)
+        if tmp:
+            read += len(tmp)
+            b.extend(tmp)
+    return bytes(b)
+
+def expect_response(sock, expected):
+    global passedAll
+    data = recv_all(s, len(expected))
+    print(data.decode('utf-8'), end='')
+    if data != expected:
+        print(f'Output mismatch\nexpected : {expected}\nreceived : {data}', file=sys.stderr)
+        passedAll = False
+        return False
+    return True
+
+def send_and_expect_response(sock, send, expected, exitOnFail=False):
+    s.sendall(send + b'\r\n')
+    print(send.decode('utf-8'))
+    if not expect_response(sock, expected + b'\r\n'):
+        if exitOnFail:
+            sys.exit(1)
+
+passedAll = True
+
+with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+    s.connect((HOST, PORT))
+
+    print("\nTesting lst")
+    send_and_expect_response(s, b'lst', b'empty')
```

Comment:
It is better to create a constant file with the frontend commands and refer those constants in these test cases. That will enable better maintenance of the code in future.

Reply:
Moved all constant strings to variables

Reply:
Ok